daml-lf/kv-support/src/test/scala/com/daml/lf/kv/transactions/TransactionTraversalSpec.scala

```diff
@@ -88,14 +47,48 @@ class TransactionTraversalSpec extends AnyFunSuite with Matchers {
   }
 
   test("extractPerPackageWitnesses - extract package witness mapping as expected") {
-    val result = TransactionTraversal.extractPerPackageWitnesses(rawTx)
+    val testTransaction = RawTransactionForTest()
+    val result = TransactionTraversal.extractPerPackageWitnesses(testTransaction.rawTx)
+    result shouldBe
+      Right(
+        Map(
+          "template_exercise" -> Set("Alice", "Charlie"),
+          "interface_exercise" -> Set("Alice", "Charlie"),
+          "template_create" -> Set("Alice", "Charlie", "Bob"),
+          "template_fetch" -> Set("Alice", "Bob"),
+        )
+      )
+  }
+
+  test(
+    "extractPerPackageWitnesses - extract package witness mapping as expected including rollback node"
+  ) {
+    val testTransaction = RawTransactionForTest()
+    val lookupByKeyNode = withNodeBuilder { builder =>
+      builder.setLookupByKey(
+        NodeLookupByKey.newBuilder
+          .setTemplateId(identifierForTemplateId("template_lookup_by_key"))
+          .setKeyWithMaintainers(
+            KeyWithMaintainers.newBuilder().addMaintainers("New maintainer")
+          )
+      )
+    }
+    testTransaction.builder.addRoot(withNodeBuilder { builder =>
```

Comment:
if you want to test your traversal logic you probably want some nodes before and some nodes after the rollback. E.g., consider a bug where in the traversal you accidentally only take the children of the rollback dropping the ones you had before in your stack. Your current test wouldn’t catch that. I think your code is fine jus tnitpicking test coverage.

Reply:
Increased the complexity of the transaction node used for rollback testing by adding some nested nodes. Ideally we should use some generators for the test data but for now this should do it.